src/core/Akka.Cluster/ClusterDaemon.cs

```diff
@@ -2204,10 +2204,9 @@ public void LeaderActionsOnConvergence()
                 _exitingConfirmed.Where(x => localGossip.GetMember(x).Status == MemberStatus.Exiting)
                 .ToImmutableHashSet();
 
+            var upNumber = 0;
```

Comment:
Possible fix for https://github.com/akkadotnet/akka.net/issues/1784 - turns out were were using `upNumber` incorrectly before and could result in multiple nodes having the same upNumber. So the aging mechanism wasn't implemented properly at all, due to what I believe is somewhat confusing syntax on the JVM: https://github.com/akka/akka/blob/master/akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala#L1079

This has been fixed. upNumber will now increment correctly. Do you think this could have resolved the cluster singleton issue @Horusiath @zbynek001 ? Any easy way to check with the existing tests?